DDD Workshop.ipynb

```diff
@@ -0,0 +1,398 @@
+    "\n",
+    "`AIRPORT,SEATS,OCCUPIED,TIME`\n",
+    "`LAX,200,180,WEEKDAY`\n",
+    "`JFK,200,161,WEEKDAY`\n",
+    "`LAX,200,186,WEEKDAY`\n",
+    "`YVR,200,0.8315980689983522,WEEKDAY`\n",
+    "`LAX,250,227,REDEYE`\n",
+    "`JFK,250,226,REDEYE`\n",
+    "`LAX,250,228,REDEYE`\n",
+    "`YVR,250,0.8663797750002538,REDEYE`\n",
+    "&vellip;\n",
+    "\n",
+    "We will access the full 1800 line file in the code, but all of the contents will follow the format above.\n",
+    "\n",
+    "\n",
+    "## Types of Conclusions\n",
+    "\n",
+    "#### What makes a high-level conclusion?\n",
+    "\n",
+    "These types of conclusions are for very broad generalizations and are mostly observations, rather than predictions. There is a way to categorize the types of insights gained from data analysis: Descriptive, Diagnostic, Predictive, and Prescriptive. These can be thought of as increasing both specificities and work required: \n",
+    "\n",
+    "![](https://i.imgur.com/HlYuOwH.png)\n",
+    "\n",
+    "<!-- cite: https://www.scnsoft.com/blog/4-types-of-data-analytics -->\n",
+    "\n",
+    "A high-level conclusion is then one of the first two of these and is broader in scope. For example, in looking at the flight data we see that there are twice as many flights to LA relative to Vancouver or New York. This information is probably easy to see, and we can confirm it over the entire dataset:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Other similar high-level conclusions might be if the size of the planes is related to the destinations if the occupancy is related to the plane sizes alone, etc. Most high-level conclusions can be drawn from the data directly, provided you are aware of the potential fallacies and have verified your data.\n",
+    "\n",
+    "Speaking of verifying your data, why is the occupied field for the YVR flights a decimal value instead of an integer number of seats like in the LAX and JFK flights? Well that would be something we have to reason about. It would seem that the Canadian flight logs record percentage of the seats occupied instead of the number of seats occupied. Not fixing this would present a barrier to analyzing the data much further, and indeed should be fixed asap:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 36,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from math import floor\n",
+    "\n",
+    "data_file = open(\"flight_data.csv\", \"r\")\n",
+    "cleaned_data = open(\"cleaned_flight_data.csv\", \"w\")\n",
+    "\n",
+    "for line in data_file:\n",
+    "    airport = line[:3]\n",
+    "    if airport == \"YVR\":\n",
+    "        fields = line.split(\",\")\n",
+    "        num_seats = int(fields[1])\n",
+    "        occ_percent = float(fields[2])\n",
+    "        occ_count = floor(num_seats * occ_percent)\n",
+    "        line = \"{},{},{},{}\".format(fields[0], fields[1], occ_count, fields[3])\n",
+    "        \n",
```

Comment:
Clarifying the programming language and adding 2-3 sentences about the code you ran would help, as this is an introduction post, so many people will not have much experience. For example, here you can describe the difference between regular and cleaned flight data, then explain what the declaration of the line variable is doing.